monkey/infection_monkey/control_channel.py

```diff
@@ -0,0 +1,52 @@
+import json
+import logging
+
+import requests
+
+from common.common_consts.timeouts import SHORT_REQUEST_TIMEOUT
+from infection_monkey.config import GUID, WormConfiguration
+from infection_monkey.control import ControlClient
+from monkey.infection_monkey.i_control_channel import IControlChannel
+
+requests.packages.urllib3.disable_warnings()
+
+logger = logging.getLogger(__name__)
+
+
+class ControlChannel(IControlChannel):
+    def should_agent_stop(self) -> bool:
+        server = WormConfiguration.current_server
+        if not server:
+            return
+
+        try:
+            response = requests.get(  # noqa: DUO123
+                f"{server}/api/monkey_control/{GUID}",
+                verify=False,
+                timeout=SHORT_REQUEST_TIMEOUT,
+            )
+
+            response = json.loads(response.content.decode())
+            return response["stop_agent"]
+        except Exception as e:
+            logger.error(f"An error occurred while trying to connect to server. {e}")
+
+    def get_config(self) -> dict:
+        return ControlClient.load_control_config()
```

Comment:
`ControlClient.load_control_config()` does not return anything. It just populates the `WormConfiguration`, so you still need to return the `WormConfiguration` from this function.

I think, ideally, we'd like this to return a `dict` and not a `Configuration` object. We should be able to use `return WormConfiguration.as_dict()`

Reply:
In what way is dict better than Configuration object?

Reply:
Today it's probably not, but in a world where we enable plugins, a dictionary is more flexible. That is, `if key in config_dict` is cleaner than `if hasattr(config_obj, key)`. While you expect dictionaries to be dynamically defined, you generally expect the attributes of an object to be statically defined.